### DISCUSSION TITLE
how to create a vertical shaded area on the bar graph

### DISCUSSION THREAD

**Discussion by plsmarann** [patrykandpatrick/vico#811] [Questions]
### Question

I am searching for a way to shade an area . eg. on the x axis form 1-4 .
<img width="427" alt="Screenshot 2024-07-24 at 11 00 04 AM" src="https://github.com/user-attachments/assets/eadccf23-ab46-469b-bf0b-105e0ee362a3">

### Vico version(s)

1.14.0

### UI framework(s)

Jetpack Compose

**Comment by plsmarann** [comment]
@patrickmichalik  any idea here

**Reply by Gowsky** [reply]
Please don't ping the maintainers. We're notified about discussions and will get back to you when we can. In the meantime, you can consider reading through related discussions and the wiki, where this question has been already answered multiple times.

**Reply by patrickmichalik** [reply]
Hello! You can implement this by means of a custom `Decoration`. Let me know if you have any questions in this regard. Discussion #775 may be helpful. We plan on adding built-in `Decoration`s for vertical boxes and lines (#615) in Vico 2.

**Comment by joseandroidengineer** [comment]
Hey @plsmarann check out the wiki, on one of the example charts provided there is a way to do this.
https://patrykandpatrick.com/vico/wiki/cartesian-charts/layers/column-layer 

If you scroll down, one of those examples may be able to help you. Perhaps this one!
https://github.com/patrykandpatrick/vico/blob/v2.0.0-alpha.27/sample/src/main/java/com/patrykandpatrick/vico/sample/showcase/charts/Chart6.kt

Hope this helps!

**Reply by Gowsky** [reply]
Please note that the question concerns creating a vertical box, while the sample chart you mention has a horizontal one.
